### PR TITLE
Gowtham Opening multiple time logs sometimes loads my time log instead

### PR DESCRIPTION
**Pull request by gowthamdongari**
# Description

image

## Related PRS (if any):
Please use the latest backend development branch.

## Main changes explained:
- Fixed the User Profile ID in the Time log entry
- Fixed the null badge count error 

## How to test:
1. check into the current branch
2. do `npm install` and `start:local` to run this PR locally
3. Clear site data/cache
4. log in as an admin user
5. go to Leaderboard→ Click on the progress Bar of all users (open it in the New Tab )
6. verify if the profile of the concerned person is shown

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/27274513/dd03ff49-be8a-4664-9a64-a436b9845cd7

## Note:
Sometimes, the pages load very slowly, and I am trying to figure out the issue. I hope this is not related to the current PR. Any suggestions are welcome.

### CODE REVIEW

**Review by Sandhya1236 — DISMISSED** [review]
I have successfully tested the recent updates which included fixing the User Profile ID in the time log entry and resolving the null badge count error. After checking into the current branch, running npm install, and starting the application locally, I cleared the site data and cache. I logged in as an admin user and navigated to the Leaderboard. By clicking on the progress bar of all users and opening it in a new tab, I verified that the profile of the concerned person was displayed correctly. All functionalities worked as expected without any issues.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/79351059/6d927af1-0658-4d73-9459-052e4ec89203

**Comment by sahithidhulipala9** [comment]
I have successfully tested the PR by logging into the admin profile, navigating to leaderboard and clicking on the users time log and opening it new tab and the profile associated is displaying correctly. 

Here is the video link: https://www.loom.com/share/296b62f0f94a449d972b401ca2ad901c?sid=f3203e5d-4c5c-48ab-a34c-e4870da2949f

**Review by sahithidhulipala9 — APPROVED** [review]

**Review by Jingyii800 — APPROVED** [review]
It worked well. I opened several user profile pages by clicking on the progress bar from admin leaderboard. It correctly the user profile page that matched the name on the leaderboard.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112589476/9a40db43-6646-4948-87e6-6a61b57bc7c4

**Review by suaniii — APPROVED** [review]
I have successfully tested the PR by logging into the admin profile, navigating to the leaderboard, and clicking on the progress bar for several users. Opening their time logs in new tabs correctly displayed the user profiles that matched the names on the leaderboard. Everything worked well.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113208952/7a738ab6-727a-4bbc-91a8-6c8c7550ed08

**Review by Sudershhh — APPROVED** [review]
Tested the PR. It is now only loading the corresponding User profile.

PR 2307

**Review by Logeshwari-Renu — APPROVED** [review]
I have tested the PR and on selecting the user under dashboard progress Bar, the respective user profile page is redirected. Verified for multiple users and it work fine!

File: src/components/SummaryBar/SummaryBar.jsx [hunk 1]
```diff
@@ -289,8 +290,8 @@ const SummaryBar = props => {
                 </font>
                 <CardTitle className={`align-middle ${darkMode ? 'text-light' : 'text-dark'}`} tag="h3">
                   <div className='font-weight-bold'>
-                    {displayUserProfile.firstName + ' '}
-                    {displayUserProfile.lastName}
+                    {userProfile?.firstName ||displayUserProfile.firstName + ' '}
+                    {userProfile?.lastName || displayUserProfile.lastName}
```
Comment (agryash) [inline src/components/SummaryBar/SummaryBar.jsx:294]: nit: space between Curly braces and the variable. ESLint gives me a warning if there's no space. Same for line 293

**Comment by agryash** [comment]
Functionality works as expected. Here's a demo for the same:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/38642926/5f92f0ab-e8a0-4288-8611-343450d9565d

File: src/components/SummaryBar/SummaryBar.jsx (hunk 1 above)
Comment (gowthamdongari) [reply src/components/SummaryBar/SummaryBar.jsx:294]: The space helps to separate first name and lastname

Comment (agryash) [reply src/components/SummaryBar/SummaryBar.jsx:294]: I meant add a space between "{" and "userProfile?.firstName".

**Review by vishavk1992 — APPROVED** [review]
I reviewed this PR while logged into my owner account. I navigated to the user profile, then to the leaderboard, and clicked on the progress bar of the user (opening it in a new tab or the current tab) . I verified that the profile of the concerned person  displayed correctly.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/148455293/93a83518-8ba1-4aa2-acc2-ed4fcc18373c

**Review by JatinAgrawal94 — APPROVED** [review]
Changes work as expected. Great work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/9be60d18-52ea-4c66-9dcd-58a8248fba6e

**Review by nyelmela — APPROVED** [review]
I tested the PR and verified that the fixes are working correctly. This includes fixing the User Profile ID in the Time log entry. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123037122/db2b52a3-d703-4568-a208-b316be6c2d4e

**Comment by one-community** [comment]
Thank you all, merging. I see a bunch of you didn't check this with multiple tabs open tough, which is what this PR was actually fixing.

**Pull request merged**